### PR TITLE
Extract Systemloaders into AutoPlugins

### PR DESCRIPTION
**Pull request by muuki88**
# Systemloader Plugin - Proposal

Goal of this PR is to refactor the `JavaServerAppPackaging` autoplugin and make system-loaders more flexible and configurable.

The main idea is to split the systemloaders in separate AutoPlugins. Usage would look like this

```scala
enablePlugins(JavaServerAppPackaging, SystemdPlugin)
```

This would trigger all the special server-folders and the configuration of a systemd start script.

Pro | Con
------------ | -------------
Systemloaders are truly isolated and can evolve independetly | No easy way to built a package with multiple systemloaders
Systemloaders can  be used with other archetypes | Possible breaking changes

Squeezing every customization into the `JavaServerAppPackaging` doesn't scale well codewise. The main pain point is, that things like this won't work anymore

```scala
serverLoading in Rpm = SystemV,
serverLoading in Debian = Systemd
```

We probably could allow something like

```scala
enablePlugins(Systemd, SystemV)
```

when break our API and really isolate the systemloaders. However I'm not sure if this is a common need to

- specify different loaders
- package with multiple loaders

The solution would, again, be submodules with different configurations.

# Related issues

- #754
- #728
- #679
- #770 

cc: @eed3si9n @dwhjames @fiadliel @metasim @kardapoltsev

### CODE REVIEW

File: src/main/scala/com/typesafe/sbt/packager/archetypes/SystemD.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import com.typesafe.sbt.packager.linux.LinuxPlugin.autoImport.serverLoading
+
+object SystemD extends AutoPlugin {
+
+  object autoImport {
+    // all systemd specific settings/tasks here
+    val systemdExitSuccessStatus = settingKey[Int]("defines the ExitSuccessStatus for systemd")
+  }
+
+  import autoImport._
```
Comment (muuki88) [inline src/main/scala/com/typesafe/sbt/packager/archetypes/SystemD.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/nepomukseiler/sbt-native-packager/file/2460298881/issues/source?bid=2040925&fileBranchId=3254858#l16)

**Comment by kardapoltsev** [comment]
It's a good idea IMHO.

**Comment by dpennell** [comment]
What would it even mean to have multiple system loaders?
- build one RPM that tries to use multiple system loaders?  
 - Does it use all available (yikes)?
 - One is default, alternates can be selected at install time?
Unless somebody steps forward with a concrete use-case, I think you should proceed.

**Comment by muuki88** [comment]
Thanks for your feedback. I will create an implementation see how everything plays out.

**Comment by muuki88** [comment]
I will merge this and release as `1.2.0-M1`. From there we can tackle all issues related to systemloader configuration issues.

**Pull request merged**